### PR TITLE
feat: can set aggregate permission

### PR DESCRIPTION
**Pull request by mswertz**
todo:
- [x] implement filter on SqlQuery so there disclosing queries are blocked (ontologies are free, otherwise count only)
- [x] implement a minimum number for count, below which the value is not shown
- [x] update graphql endpoint to remove api not available to aggregate so easier not to get errors
- [x] migration for row level policy on metadata (aggregator instead of viewer)
- [x] can assign aggregator role via user interface
- [x] in table explorer only show aggregates for tables having aggregate permission and grouby for ontologies
- [x] in download view remove non-ontology downloads 
- [x] check if downloads don't include sensitive data and don't error
- [x] can only filter on ontology terms (!)
- [x] how to display the '10 or less' values?
- [x] migration of new roles to be added for all existing schemas

probably in next PR:
- [ ] deal with other types of aggregation, i.e., 'exists' and 'range'
- [ ] allow configuration of the minimum number

### CODE REVIEW

File: apps/molgenis-components/src/components/tables/AggregateTable.vue [hunk 1]
```diff
@@ -191,9 +192,9 @@ export default defineComponent({
       if (this.refColumns?.length > 0) {
         this.selectedColumn = this.refColumns[0];
         this.selectedRow = this.refColumns[1] || this.refColumns[0];
+        //this.fetchData();
```
Comment (chinook25) [inline apps/molgenis-components/src/components/tables/AggregateTable.vue:195]: remove

**Review by chinook25 — COMMENTED** [review]
Just some minor things.

File: apps/molgenis-components/src/components/tables/AggregateTable.vue (hunk 1 above)
Comment (mswertz) [reply apps/molgenis-components/src/components/tables/AggregateTable.vue:195]: this is the place where my issue with massive reloading exists and where I might need help

File: backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/SqlQuery.java [hunk 2]
```diff
@@ -136,6 +140,13 @@ public List<Row> retrieveRows() {
     }
   }
 
+  private void checkHasViewPermission(SqlTableMetadata table) {
+    if (!table.getTableType().equals(TableType.ONTOLOGIES)
+        && !schema.getInheritedRolesForActiveUser().contains(VIEWER.toString())) {
```
[the remark this answers is not in the document]
Comment (mswertz) [reply backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/SqlQuery.java:145]: indeed, will doe

File: apps/nuxt3-ssr/utils/fingerprint.js [hunk 3]
```diff
@@ -1 +1 @@
-export const hash = "";
+export const hash = "lj5cwkhj";
```
[the remark this answers is not in the document]
Comment (mswertz) [reply apps/nuxt3-ssr/utils/fingerprint.js:1]: darn, this is so annoying. Can we ignore this somehow?

[the remark this answers is not in the document]
Comment (chinook25) [reply apps/nuxt3-ssr/utils/fingerprint.js:1]: it's already in the .gitignore of that folder, maybe it needs to be added to other .gitignore's as well?

File: apps/dev-proxy.config.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
-const HOST = process.env.MOLGENIS_APPS_HOST || "https://emx2.dev.molgenis.org";
+const HOST = process.env.MOLGENIS_APPS_HOST || "http://localhost:8080";
```
Comment (mswertz) [inline apps/dev-proxy.config.js:1]: should be undone

File: backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/SqlQuery.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (mswertz) [reply backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/SqlQuery.java:145]: well, is actually making things worse so reverted

**Review by chinook25 — CHANGES_REQUESTED** [review]
- I'm missing a readme/manual.
- After running this branch all my public schemas where no longer showing when not logged in.
-Cleaning my dbs after this throws error, probably due to missing CASCADES (Caused by: org.postgresql.util.PSQLException: ERROR: role "MG_ROLE_pg_toast/Aggregator" cannot be dropped because some objects depend on it). 
 - After that, and restarting the backend I don't see the schema's but I also cannot log in as admin anymore.

**Comment by mswertz** [comment]
> * After that, and restarting the backend I don't see the schema's but I also cannot log in as admin anymore.

did you try to run 'gradle cleandb'?

File: apps/molgenis-components/src/components/tables/AggregateTable.vue [hunk 5]
```diff
@@ -185,30 +187,38 @@ export default defineComponent({
...
+          this.canView
+        );
+      }
       if (this.refColumns?.length > 0) {
         this.selectedColumn = this.refColumns[0];
         this.selectedRow = this.refColumns[1] || this.refColumns[0];
       }
-    }
-    this.fetchData();
+      this.fetchData();
```
[the remark this answers is not in the document]
Comment (mswertz) [reply apps/molgenis-components/src/components/tables/AggregateTable.vue:208]: I think this diff is deceiving as if a '}' was deleted, it is actually only once run on update of 'allColumns

File: apps/tables/src/components/ViewTable.vue [hunk 6]
```diff
@@ -25,6 +26,20 @@ export default {
     RoutedTableExplorer,
   },
   computed: {
+    canView() {
+      if (this.session) {
```
Comment (jelmerveen) [inline apps/tables/src/components/ViewTable.vue:30]: anonymous users have no session?

Comment (mswertz) [reply apps/tables/src/components/ViewTable.vue:30]: jawel, maar is wellicht nog niet geladen. Maar dat kan vast nog mooier.

Comment (mswertz) [reply apps/tables/src/components/ViewTable.vue:30]: nog ff gekeken maar dit is denk ik zoals het moet. Het is reactive dus zou opnieuw moeten vuren als session verandert toch?

**Pull request merged**